src/targets/gpu/target.cpp

```diff
@@ -72,6 +73,7 @@ std::vector<pass> target::get_passes(migraphx::context& gctx) const
         write_literals{&ctx},
         schedule{gpu::schedule_model{ctx.get_current_device().nstreams()}, not enabled(MIGRAPHX_DISABLE_SCHEDULE_PASS{})},
         memory_coloring{"hip::allocate"},
+        preallocate_param{"scratch", &ctx},
```

Comment:
Does this mean scratch memory is always automatically allocated? 
Do we need providing an option for user to allocate the scratch?